Testnet/Multisig/Multisig/MultisigContract.cs

```diff
@@ -0,0 +1,355 @@
+    {
+        get => State.GetUInt64(nameof(TransactionCount));
+        private set => State.SetUInt64(nameof(TransactionCount), value);
+    }
+
+    public bool IsOwner(Address address)
+    {
+        return State.GetBool($"{OwnerPrefix}:{address}");
+    }
+
+    public bool IsConfirmedBy(ulong transactionId, Address address)
+    {
+        return State.GetBool($"{ConfirmationPrefix}:{transactionId}:{address}");
+    }
+
+    public Transaction GetTransaction(ulong transactionId)
+    {
+        Assert(transactionId <= TransactionCount, "Can't retrieve a transaction that doesn't exist yet.");
+
+        return State.GetStruct<Transaction>($"{TransactionPrefix}:{transactionId}");
+    }
+    
+    public void AddOwner(Address address)
+    {
+        EnsureWalletOnly();
+
+        Assert(!IsOwner(address), "Can't add an existing owner.");
+
+        SetOwner(address);
+
+        OwnerCount++;
+    }
+
+    public void RemoveOwner(Address address)
+    {
+        EnsureWalletOnly();
+
+        Assert(IsOwner(address), "Can't remove a non-owner.");
+        Assert(OwnerCount > Required, "Can't have less owners than the minimum confirmation requirement.");
+
+        UnsetOwner(address);
+
+        OwnerCount--;
+    }
+
+    /// <summary>
+    /// Only called within the contract constructor to initialise the owners.
+    /// </summary>
+    /// <param name="addresses">The array of addresses serialised as bytes.</param>
+    private void SetOwners(byte[] addresses)
+    {
+        Address[] addressList = Serializer.ToArray<Address>(addresses);
+
+        foreach (var address in addressList)
+        {
+            SetOwner(address);
+        }
+
+        OwnerCount = (uint)addressList.Length;
+    }
+
+    private void SetOwner(Address address)
+    {
+        State.SetBool($"{OwnerPrefix}:{address}", true);
+
+        Log(new OwnerAddition() { Owner = address });
+    }
+
+    private void UnsetOwner(Address address)
+    {
+        State.SetBool($"{OwnerPrefix}:{address}", false);
+
+        Log(new OwnerRemoval() { Owner = address });
+    }
+
+    /// <summary>
+    /// Submits method and parameter metadata describing a contract call that the multisig contract should invoke once it is sufficiently confirmed.
+    /// </summary>
+    /// <param name="data">An array of method parameters encoded as packed byte arrays. See the <see cref="Transaction"/> struct for further information.</param>
+    /// <returns>The transactionId of the submitted multisig transaction.</returns>
+    /// <remarks>The submitter implicitly provides a confirmation for the submitted transaction.</remarks>
+    public ulong Submit(Address destination, string methodName, byte[] data)
+    {
+        EnsureOwnersOnly();
+
+        TransactionCount++;
+
+        State.SetStruct($"{TransactionPrefix}:{TransactionCount}", new Transaction()
+        {
+            Destination = destination,
+            Executed = false,
+            Value = 0,
```

Comment:
Am I right in thinking this only supports transactions which require not sending CRS?

Could potentially support owners paying by having a target and current transaction value, as well as a block deadline to refund failed transactions. Or slightly more simple might be to have the submitter pay.